File: Bit Manipulation/67-Add-Binary.java
// import java.util.StringBuild;

class Solution {
    public String addBinary(String a, String b) {
        StringBuilder s = new StringBuilder();
        int i = a.length() - 1; int j = b.length() - 1; int carry = 0;        
        while (i >= 0 || j >= 0) {
            int sum = carry;
            if (i >= 0) {sum += a.charAt(i--) - '0';}
            if (j >= 0) {sum += b.charAt(i--) - '0';}
            s.append(sum == 2 || sum == 0 ? 0 : 1);
            carry = sum == 2 || sum == 0 ? 0 : 1;
        }
        if (carry != 0) {
            s.append(carry);
        }
        return s.reverse().toString();
    }
}

File: Bit Manipulation/67-Add-Binary/67-Add-Binary.ts
function addBinary(a: string, b: string): string {
    let str: string[] = [];
    let i = a.length - 1; let j = b.length - 1; let carry = 0;

    while (i >= 0 || j >= 0) {
        let sum = carry;
        if (i >= 0) {sum += Number(a[i--]) - 0;}
        if (j >= 0) {sum += Number(b[j--]) - 0;}
        str.push((sum % 2).toString());
        carry = Math.floor(sum / 2);
    }

    if (carry != 0) {
        str.push("1");
    }

    return str.reverse().join("");
};